src/puppetlabs/puppetdb/command.clj

```diff
@@ -86,23 +101,78 @@
    "store report" (version-range 3 6)
    "deactivate node" (version-range 1 3)})
 
-;; ## Command parsing
+(defn- read-command-body
+  [body]
+  (if (= (class body) (class (byte-array 0)))
```

Comment:
would be nice to factor this out into a `byte-array?` fn, maybe something like...

```clojure
(def byte-array?
  (let [t (Class/forName "[B")]
    (fn [x]
      (instance? t x))))
```